ibmcloud/cluster/README.md

```diff
@@ -0,0 +1,165 @@
+This can then be used to test the cloud-api adaptor and peer-pods.
+
+## Prerequisites
+
+To create the Kubernetes cluster, you need to install terraform, Ansible and the IBM Cloud CLI on your 
+'development machine'. To manage the cluster you will need to install `kubectl`.
+Please follow the the official installation guides:
+
+* [Install Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+
+> **Tip:** If you are using Ubuntu linux, you can run follow commands simply:
+> ```bash
+> $ sudo apt-get update && sudo apt-get install -y gnupg software-properties-common curl
+> $ curl -fsSL https://apt.releases.hashicorp.com/gpg | sudo apt-key add -
+> $ sudo apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main"
+> $ sudo apt-get install terraform -y
+> ```
+
+* [Install Ansible](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html)
+
+> **Tip:** If you are using Ubuntu linux, you can run follow commands simply:
+> ```bash
+> $ sudo apt-get install -y python3
+> $ sudo ln -s /usr/bin/python3 /usr/bin/python
+> $ sudo add-apt-repository --yes --update ppa:ansible/ansible
+> $ sudo apt-get install ansible -y
+> ```
+
+* [Installing the stand-alone IBM Cloud CLI](https://cloud.ibm.com/docs/cli?topic=cli-install-ibmcloud-cli)
+
+> **Tips**
+> - If you are using Ubuntu linux, you can run follow commands simply:
+>     ```bash
+>     $ curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+>     $ ibmcloud plugin install vpc-infrastructure
+>     $ ibmcloud plugin install cloud-object-storage
+>     ```
+> - You can use the [IBM Cloud Web UI](https://cloud.ibm.com/vpc-ext/overview) for most of the operations of IBM Cloud.
+And please make sure that you are selecting the correct region in the Web UI.
+> 
+
+* You need IBM Cloud API key. You can create your own API key at [https://cloud.ibm.com/iam/apikeys](https://cloud.ibm.com/iam/apikeys).
+
+* [Install `kubectl` to run commands against the Kubernetes cluster](https://kubernetes.io/docs/tasks/tools/)
+
+## Create an IBM Cloud Virtual Private Cloud (VPC) and 'self-managed' Kubernetes Cluster -
+
+The Terraform configuration supports building the cluster on both the x86 (Intel) and s390x (IBM Z) architectures.
+
+To use the Terraform configuration, you need to create a file `terraform.tfvars` to specify parameters for the
+Terraform configuration. The `terraform.tfvars` file with all mandatory parameters looks like this:
+
+```
+ibmcloud_api_key = "<your API key>"
+ssh_key_name = "<name of your IBM Cloud create SSH key>"
+```
+
+There are also a number of optional fields:
+```
+cluster_name = "<name of your cluster>
+region_name = "<name of an IBM Cloud region>"
+zone_name = "<name of a zone in your IBM Cloud zone region>"
+ssh_pub_key = "<your SSH public key - this results in the ssh_key being created on IBM Cloud>"
```

Comment:
Shall we explain this key pair will be used for terraform to ssh to the node to perform kubeadm operations?

Reply:
yes I think that is useful to add, particularly that this needs to be password-less and on the host running the script. @stevenhorsman would you mind adding that?

Reply:
I've added some extra details in the Parameters section. I hope that helps?

Reply:
Thanks for the fix!